### PR TITLE
Update: Add autofix for `no-extra-label` (fixes #5952)

### LINKED ISSUE #5952: Implement autofix for `no-extra-label`
I think this is safe to add, I'm working on this.

### PR DESCRIPTION
**Pull request by alberto**

### CODE REVIEW

File: lib/rules/no-extra-label.js [hunk 1]
```diff
@@ -105,10 +120,14 @@ module.exports = {
             while (info) {
                 if (info.breakable || info.label === label) {
                     if (info.breakable && info.label === label) {
+                        var prevToken = sourceCode.getTokenOrCommentBefore(node.label);
+                        var range = [prevToken.range[1], node.label.range[1]];
+
                         context.report({
-                            node: labelNode,
+                            node: node.label,
```
Comment (mysticatea) [inline lib/rules/no-extra-label.js]: Why do you change this?

Comment (mysticatea) [inline lib/rules/no-extra-label.js]: Or all `labelNode` can be removed with replacing `node.label`.

Comment (alberto) [inline lib/rules/no-extra-label.js]: Just an oversight on my part. 👍

File: tests/lib/rules/no-extra-label.js [hunk 2]
```diff
@@ -39,15 +39,60 @@ ruleTester.run("no-extra-label", rule, {
...
+            errors: ["This label 'A' is unnecessary."]
+        },
+        {
+            code: "X: while (x) { A: while (a) { B: { break A; break B; continue X; } } }",
+            output: "X: while (x) { A: while (a) { B: { break; break B; continue X; } } }",
+            errors: ["This label 'A' is unnecessary."]
+        },
+        {
+            code: "A: do { break A; } while (a);",
+            output: "A: do { break; } while (a);",
```
Comment (nzakas) [inline tests/lib/rules/no-extra-label.js]: Why not remove the label, too?

Comment (alberto) [inline tests/lib/rules/no-extra-label.js]: That was covered in #5955. I'm closing this, since we have the same concern about typos as there.

**Pull request closed without merging**